src/main/java/org/commcare/formplayer/services/EntitiesSelectionService.java

```diff
@@ -0,0 +1,72 @@
+package org.commcare.formplayer.services;
+
+import com.timgroup.statsd.StatsDClient;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.commcare.core.interfaces.EntitiesSelectionCache;
+import org.commcare.formplayer.objects.EntitiesSelection;
+import org.commcare.formplayer.repo.EntitiesSelectionRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.CacheConfig;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.annotation.Nullable;
+
+@Service
+@CacheConfig(cacheNames = {"entities_selection"})
+public class EntitiesSelectionService implements EntitiesSelectionCache {
+
+    private final Log log = LogFactory.getLog(EntitiesSelectionService.class);
+
+    @Autowired
+    private EntitiesSelectionRepo entitiesSelectionRepo;
+
+    @Autowired
+    private FormplayerStorageFactory storageFactory;
+
+    @Autowired(required = false)
+    private StatsDClient datadogStatsDClient;
+
+    @Autowired
+    CacheManager cacheManager;
+
+    @Override
+    public UUID write(String[] values) {
+        EntitiesSelection entitySelection =  new EntitiesSelection(storageFactory.getUsername(),
+                storageFactory.getDomain(), storageFactory.getAppId(), storageFactory.getAsUsername(), values);
+        EntitiesSelection entitiesSelection = entitiesSelectionRepo.save(entitySelection);
+        Cache cache = cacheManager.getCache("entities_selection");
+        cache.put(entitiesSelection.getId(), entitiesSelection.getEntities());
+        return entitiesSelection.getId();
+    }
+
+    @Nullable
+    @Override
+    @Cacheable
+    public String[] read(UUID key) {
```

Comment:
this should validate that the values found belong to the current user

Reply:
decided to [implement the validation](https://github.com/dimagi/formplayer/pull/1118/commits/318cca9530133616045156802a946476f3ed0beb) against username, appid pair as same user can exist through multiple apps